extensions/meme_rater.py

```diff
@@ -0,0 +1,75 @@
+import os
+import logging
+import hikari
+import lightbulb
+from PIL import Image
+import requests
+from extensions.snark import model
+from google.generativeai.generative_models import GenerativeModel
+from typing import Final
+
+plugin = lightbulb.Plugin("MemeRater")
+
+MEME_CHANNEL_ID = int(os.environ.get("MEME_CHANNEL_ID", 0))
+MEME_RATE_PROMPT: Final[str] = os.environ.get("MEME_RATING_PROMPT",
+    "Rate this meme out of 10, with 10 being the funniest. Rate is solely on how funny a bunch of computer science nerds would find it. ONLY Return an integer."
+)
+MINIMUM_MEME_RATING_TO_NOT_DELETE: Final[int] = int(os.environ.get("MEME_QUALITY_THRESHOLD", "6"))
+DELETE_SHIT: Final[bool] = False
+IMG_FILE_EXTENSIONS: Final = {"jpg", "jpeg", "png", "webp"}
+
+
+def get_meme_rating(image_url: str, model: GenerativeModel):
+    image = requests.get(image_url, stream=True).raw
+    if not image:
+        return ''
+    response = model.generate_content(
+        contents=[MEME_RATE_PROMPT, Image.open(image)]
+    )
+    logging.info(f'Meme rating response: {response.text}')
+    return response.text
+
+def number_emoji(number: int):
+    if number == 10:
+        unicode = "\N{KEYCAP TEN}"
+    else:
+        unicode = f"{number}\N{VARIATION SELECTOR-16}\N{COMBINING ENCLOSING KEYCAP}"
+    return hikari.UnicodeEmoji.parse(unicode)
+
+@plugin.listener(hikari.GuildMessageCreateEvent)
+async def main(event: hikari.GuildMessageCreateEvent) -> None:
+    if event.channel_id != MEME_CHANNEL_ID:
+        return
+
+    ratings = []
+    for attachment in event.message.attachments:
+        att_ext = attachment.extension
+        if att_ext not in IMG_FILE_EXTENSIONS:
+            continue
+        image_url = attachment.url
+        rating = get_meme_rating(image_url, model)
+        try:
+            ratings.append(int(rating))
+        except ValueError:
+            continue
+    if not ratings:
+        return
+
+    avg_rating = min(max(0, sum(ratings) // len(ratings)), 10)
+
+    await event.message.add_reaction(emoji=number_emoji(avg_rating))
+    await event.message.add_reaction(emoji="🐱")
+
+    if avg_rating >= MINIMUM_MEME_RATING_TO_NOT_DELETE:
+        await event.message.add_reaction(emoji="👍")
+    elif DELETE_SHIT:
+        await event.message.delete()
+        await event.message.respond(
```

Comment:
Can you respond post-deletion?

Reply:
I can swap the order of these two lines around if it'll make you feel better but it won't change the observable result.